### PR TITLE
Flutter for iOS - Views

### PR DESCRIPTION
**Pull request by niamh-power**
This PR is for the first section of the work to create the `flutter-for-ios.md` document. Reopened with correct email for CLA.

Following the same rough format as the `flutter-for-android.md` document, with iOS details instead.

The following sections have been created:
* What is the equivalent of a View in Flutter?
* How do I update widgets?
* How do I layout my widgets? Where are my storyboards?
* How do I add or remove a component from my layout?
* How do I animate a Widget?
* How do I use a Canvas to draw/paint?
* How do I build custom widgets?

### CODE REVIEW

**Comment by blundell** [comment]
I signed it!

**Comment by blundell** [comment]
I signed it! (and it's now signed at your end :) )

File: flutter-for-ios.md [hunk 1]
```diff
@@ -27,3 +27,429 @@ that are most relevant to your needs.
...
+containers for other `UIView` classes, which form your layout.
+
+In Flutter, the equivalent to this is a `Widget`. However, these have a few
+differences to a  `UIView`. To start, widgets only exist for one frame, and on
+every new frame, Flutter’s framework creates a new tree of widget instances.
+In comparison, an iOS view is drawn once and does not redraw until it is
+invalidated using `setNeedsDisplay()`.
+
+Furthermore, unlike `UIView`, Flutter’s widgets are immutable, which allows
+them to be very lightweight. By default, Flutter’s widgets are styled as
```
Comment (mjohnsullivan) [inline flutter-for-ios.md]: Not by default, only if you select the MaterialApp and use the Material widgets. You can start with Cupertino if you want. Think about wording as we don't want to give the impression that we're iOS styling unfriendly.

File: flutter-for-ios.md [hunk 2]
```diff
@@ -27,3 +27,429 @@ that are most relevant to your needs.
...
+your logo as the `image`. If the logo is not going to change during runtime,
+you would use a `StatelessWidget` in Flutter.
+
+If you want to dynamically change the UI based on data received after making an
+HTTP call then you would use a `StatefulWidget`. After the HTTP call had
+complete, you would tell the Flutter framework that the widget’s `State` has
+been updated so it could update the widget.
+
+The important thing to note here is at the core both stateless and stateful
+widgets behave the same i.e. they rebuild every frame. The difference is that
```
Comment (mjohnsullivan) [inline flutter-for-ios.md]: Only if necessary. Flutter doesn't run the build() method on every frame. Only when it needs to redraw the widget.

File: flutter-for-ios.md [hunk 3]
```diff
@@ -27,3 +27,429 @@ that are most relevant to your needs.
...
+the `StatefulWidget` has a `State` object which stores state data across
+frames and restores it after being rebuilt.
+
+If you are in doubt, then always remember this rule: if a widget changes (e.g.
+because of runtime user interactions) it’s stateful. If the only time data
+changes is at instantiation, it’s stateless. However, if a widget is reacting
```
Comment (mjohnsullivan) [inline flutter-for-ios.md]: Confusing sentence. Think about rewording.

File: flutter-for-ios.md [hunk 4]
```diff
@@ -27,3 +27,429 @@ that are most relevant to your needs.
...
+
+In iOS, you would create an animation by calling the
+`animate(withDuration:animations:)` method on a view. In Flutter, animating
+widgets can be done via the animation library by wrapping widgets inside an
+animated widget.
+
+In Flutter you use an `AnimationController` which is an `Animation<double>`
+that can pause, seek, stop and reverse the animation. It requires a `Ticker`
+which signals when vsync happens and produces a linear interpolation between
+0 and 1 on each frame while it's running.You then create one or more
```
Comment (mjohnsullivan) [inline flutter-for-ios.md]: Spacing around full stop

**Comment by Hixie** [comment]
Looks like there are two different GitHub users with commits on this PR; both need to be associated with accounts that have signed the CLA.

File: flutter-for-ios.md [hunk 5]
```diff
@@ -27,3 +27,429 @@ that are most relevant to your needs.
...
+        onPressed: _updateText,
+        tooltip: 'Update Text',
+        child: new Icon(Icons.update),
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+## How do I layout my widgets? Where is my Storyboard?
```
Comment (Sfshaza) [inline flutter-for-ios.md]: In this case, it's "How do I lay out..."

**Comment by Hixie** [comment]
What is the status of this PR?

**Comment by Sfshaza** [comment]
@ian, it looks like I'll have to make sure every bit of feedback from @mjohnsullivan was incorporated, but this PR won't be merged.

**Comment by Sfshaza** [comment]
I've incorporated all of Matt's feedback in PR https://github.com/flutter/website/pull/1046.

Closing.

**Pull request closed without merging**